internal/gatewayapi/backendtrafficpolicy.go

```diff
@@ -509,6 +517,12 @@ func (t *Translator) translateBackendTrafficPolicyForGateway(
 
 	ds = translateDNS(policy.Spec.ClusterSettings)
 
+	// Create BackendTrafficPolicy instance
```

Comment:
the IR structs use a `name` field as a differentiator https://github.com/envoyproxy/gateway/blob/main/internal/ir/xds.go
can we use that same style here ?

it can be generated with a helper similar to https://github.com/envoyproxy/gateway/blob/0c22094563ee1462223c97e867865c613b302b16/internal/gatewayapi/helpers.go#L394